### ISSUE TITLE
ItemsRepeater treats invisible items incorrectly

### ISSUE THREAD

**Issue by FoggyFinder** [AvaloniaUI/Avalonia#4453]
MCVE

xaml

```xml
<ScrollViewer VerticalScrollBarVisibility="Visible">
  <ScrollViewer.Resources>
    <local:SomeConverter x:Key="i2v" />
  </ScrollViewer.Resources>
  <ItemsRepeater Items="{Binding Items}">
    <ItemsRepeater.ItemTemplate>
      <DataTemplate>
        <Grid IsVisible="{Binding ., Converter={StaticResource i2v}}">
          <TextBlock Text="{Binding .}" />
        </Grid>
      </DataTemplate>
    </ItemsRepeater.ItemTemplate>
  </ItemsRepeater>
</ScrollViewer>
```

C#

```cs
public class Context
{
    public IEnumerable<int> Items { get; }
    public Context()
    {
        Items = Enumerable.Range(0, 100).Select(v => v).ToArray();
    }
}

public class SomeConverter : IValueConverter
{
    public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
    {
        return value switch
        {
            int v => v % 2 == 0,
            _ => false
        };
    }

    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
    {
        throw new NotImplementedException();
    }
}
```

Result:

![ir2](https://user-images.githubusercontent.com/16510738/89648115-60106480-d8c7-11ea-9efc-5df43bd1661a.gif)

Items are getting eaten

**Comment by grokys** [comment]
I'm not sure `ItemsRepeater` handles making items invisible. This is something that will need to be tested on the upstream WinUI control.

**Comment by grokys** [comment]
Fixed by #10864 and #11127.
